### DISCUSSION TITLE
More than one published version for a content at the same time?

### DISCUSSION THREAD

**Discussion by HengzheLi** [OrchardCMS/OrchardCore#10775] [General]
I just walk through the source code of DefaultContentManager.cs then stop at https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs#L328. 
I got a question about this code line. It saves the previous version of the content than change the previous content to unpublish but not save the unpublished content again. Does it mean there will be more than one published version in the database at the same time. In my understand, it should only have one published version at a time and the other versions should be unpublished. So I think the previous version of the content should be changed to unpublished before its saved. Like
```
if (previous != null)
{
    previous.Published = false;
    _session.Save(previous, checkConcurrency: true);
}
```

Does it a bug here?

**Comment by ns8482e** [comment]
Is it just your observation based on the code above OR  Do you really have two published version in db?

**Comment by jtkech** [comment]
@HengzheLi 

When doing `_session.Save()` the related entity is not yet persisted, only when doing `.SaveChangesAsync()` at the end of the current scope / transaction. So that's okay to do a mutation just after by referencing the same entity.

**Reply by HengzheLi** [reply]
Thanks for the explaination.
